test_embark/locking.js

```diff
@@ -0,0 +1,405 @@
+const { assertRevert } = require('./helpers/assertThrow')
+const { getEvent } = require('./helpers/getEvent')
+
+const Staking = embark.require('Embark/contracts/Staking');
+const StandardTokenMock = embark.require('Embark/contracts/StandardTokenMock');
+const LockManagerMock = embark.require('Embark/contracts/LockManagerMock');
+
+let accounts
+
+config({}, (err, accts) => {accounts = accts})
+
+contract('Staking app, Locking', () => {
+  let staking, token, lockManager, stakingAddress, tokenAddress, lockManagerAddress
+  let owner, user1, user2
+
+  const zeroBytes = "0x00"
+  const MAX_UINT64 = (new web3.utils.BN(2)).pow(new web3.utils.BN(64)).sub(new web3.utils.BN(1))
+  const defaultAmount = 120
+
+  const approveAndStake = async(amount = defaultAmount, from = owner) => {
+    await token.approve(stakingAddress, amount).send({ from: from })
+    await staking.stake(amount, web3.utils.asciiToHex('')).send({ from: from })
+  }
+
+  const approveStakeAndLock = async(
+    manager,
+    lockAmount = defaultAmount / 2,
+    stakeAmount = defaultAmount,
+    from = owner
+  ) => {
+    await approveAndStake(stakeAmount, from)
+    const r = await staking.lock(lockAmount, manager, web3.utils.asciiToHex('')).send({ from: from })
+    const lockId = getEvent(r, 'Locked', 'lockId')
+
+    return lockId
+  }
+
+  before(async () => {
+    owner = accounts[0]
+    user1 = accounts[1]
+    user2 = accounts[2]
+  })
+
+  beforeEach(async () => {
+    const initialAmount = 1000 * defaultAmount
+    const tokenContract = await StandardTokenMock.deploy({arguments: [owner, initialAmount]}).send()
+    token = tokenContract.methods
+    tokenAddress = tokenContract.options.address
+    const stakingContract = await Staking.deploy({arguments: [tokenAddress]}).send()
+    staking = stakingContract.methods
+    stakingAddress = stakingContract.options.address
+    const lockManagerContract = await LockManagerMock.deploy().send()
+    lockManager = lockManagerContract.methods
+    lockManagerAddress = lockManagerContract.options.address
+  })
+
+  it('locks', async () => {
+    const lockId = await approveStakeAndLock(user1)
+
+    // check lock values
+    const lock = await staking.getLock(owner, lockId).call()
```

Comment:
Better to use this syntax: either `const [ amount, time, ...] = await staking.getLock(...)` or `const [ amount, time, ...] = lock`